numba/tests/test_builtins.py

```diff
@@ -505,6 +508,24 @@ def test_hex_npm(self):
         with self.assertTypingError():
             self.test_hex(flags=no_pyobj_flags)
 
+    def test_str(self, flags=nrt_no_pyobj_flags):
+        pyfunc = str_usecase
+
+        cr = compile_isolated(pyfunc, (types.int64,), flags=flags)
+        cfunc = cr.entry_point
+
+        args = [
+            123456789,
+            2**32-1,
+            0,
+            33,
+            -33,
```

Comment:
Probably ought to test other things that are integers, e.g. `np.uint64` /

Reply:
Done!

Reply:
The change made isn't actually testing the types presented, the compile result is for a `(types.int64,)` only, all inputs will just end up being cast.

Reply:
Is the expected behaviour a cast to float64 in the operation `n % 10`, where `n` is an integer?

Reply:
No, this is a bug:
```python
>>> type(5 % 10)
<class 'int'>
>>> import numpy as np
>>> type(np.int64(5) % np.int64(10))
<class 'numpy.int64'>
```

Reply:
Well, it doesn't cast to `float64`:
```python
>>> import numba as nb
>>> @nb.njit
... def mod(a, b):
...     return a % b
...
>>> type(mod(5, 10))
<class 'int'>
```

Reply:
Is there a minimal working reproducer for what is being discussed here in relation to cast to float64, if so I can take a look? Thanks.

Reply:
Try this:
```python
from numba import njit, types
import numpy as np

@njit
def foo(n):
    l = []
    while n > 0:
        c = chr(ord('0') + (n % 10))
        n = n // 10
        l.append(c)
    return ''.join(l[::-1])

x = foo(types.ulonglong(123))
print(x)

```

Reply:
Thanks, the issue is that `n = n // 10` is doing an unsigned/signed module which results in a float64. It's following NumPy semantics, e.g.:
```
In [45]: np.uint64(11)//np.int64(10)                                                                            
Out[45]: 1.0

In [46]: type(np.uint64(11)//np.int64(10))                                                                                      
Out[46]: numpy.float64
```

Reply:
https://github.com/numba/numba/pull/5463#discussion_r452976222 should address this.